### PR TITLE
WRQ-22937: Updated storybook to version 8

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Storybook needed to be updated to the latest version

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Updated list of storybook dependencies and their version to v8.x based on the migration guide https://github.com/storybookjs/storybook/blob/next/MIGRATION.md
Changed configuration based on the storybook v8 requirements 

- Storybook 7.0 requires Node 18 or above
- removed custom docs addon because starting with storybook 7 doesn't allow customization of its addon/docs https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#dropped-addon-docs-manual-configuration
- removed source-loader configuration since it was dropped by storybook 7. I compared the docs ui of branch feature/storybook8 with the one on develop branch and it looks the same

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRQ-22937

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian ([daniel.stoian@lgepartner.com](mailto:daniel.stoian@lgepartner.com))

### CODE REVIEW

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -27,10 +27,9 @@
   },
   "dependencies": {
     "@enact/dev-utils": "^6.1.0",
-    "@storybook/addon-actions": "^6.5.16",
-    "@storybook/addon-controls": "^6.5.16",
-    "@storybook/addon-docs": "^6.5.16",
-    "@storybook/addon-toolbars": "^6.5.16",
+    "@storybook/addon-actions": "^8.1.4",
```
Comment (MikyungKim) [inline package.json:30]: Storybook just released 8.1.6 version :( Would you please update? I'm about to merge this PR.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Comment by daniel-stoian-lgp** [comment]
> @daniel-stoian-lgp @daniel-stoian-lgp Would you pleae double check the site icon? Icon shows storybook icon image

@seunghoh  Ok, I will investigate it

**Pull request merged**